2_process/src/lake_sequences_mntoha.py

```diff
@@ -0,0 +1,241 @@
+import numpy as np
+import pandas as pd
+
+def all_dates_depths(lake_obs_interpolated, 
+                     depths,
+                     temp_col='temp',
+                     depth_col='interpolated_depth',
+                     date_col='date',
```

Comment:
Okay parting thought - I've been reflecting on it, and if these 3 column name parameters could change with different data sources, it feels a little brittle to have them hidden as default arguments. I'm not sure if this would violate `snakemake` best practices, but it would feel more in keeping with our standard pipeline practices of documenting all key parameters and decisions in the main 'recipe' file if these columns were specified as `params` to the `rule mntoha_lake_sequences` rule, then passed through as arguments to `assemble_lake_data()` and then as arguments to `all_dates_depth()`.

Making that change here may not be warranted/may not make sense, but as you keep developing the pipeline, I think it would be worth exploring specifying more arguments with `params`, like you ended up doing with the sb_ids in `1_fetch.smk`. I think it would make the `snakemake` structure a bit closer to the structure of our `targets` pipelines.

Reply:
That sounds good. I like the idea of capturing them in the recipe file.